trieste/models/gpflux/models.py

```diff
@@ -143,10 +144,17 @@ def optimize(self, dataset: Dataset) -> None:
         Optimize the model with the specified `dataset`.
         :param dataset: The data with which to optimize the `model`.
         """
-        self.model_keras.fit(
-            {"inputs": dataset.query_points, "targets": dataset.observations}, **self._fit_args
+        fit_args = self._fit_args
```

Comment:
This will currently also override the entry in the original self._fit_args, which I don't think is what you want?

```suggestion
        fit_args = dict(self._fit_args)
```